packages/@aws-cdk-testing/framework-integ/test/aws-elasticloadbalancingv2/test/integ.alb.dualstack-without-public-ipv4.ts

```diff
@@ -0,0 +1,127 @@
+ *
+ */
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-cdk-elbv2-integ-dualstack-without-public-ipv4');
+
+const vpc = new ec2.Vpc(stack, 'VPC', {
+  restrictDefaultSecurityGroup: false,
+  maxAzs: 2,
+});
+
+const ipv6Block = new ec2.CfnVPCCidrBlock(
+  stack,
+  'IPv6_Block',
+  {
+    vpcId: vpc.vpcId,
+    amazonProvidedIpv6CidrBlock: true,
+  },
+);
+
+// Get the vpc's internet gateway so we can create default routes for the
+// public subnets.
+const internetGateway = valueOrDie<IConstruct, ec2.CfnInternetGateway>(
+  vpc.node.children.find(c => c instanceof ec2.CfnInternetGateway),
+  new Error('Couldn\'t find an internet gateway'),
+);
+
+const lb = new elbv2.ApplicationLoadBalancer(stack, 'LB', {
+  vpc,
+  ipAddressType: elbv2.IpAddressType.DUAL_STACK_WITHOUT_PUBLIC_IPV4,
+  internetFacing: true,
+});
+
+const listener = lb.addListener('Listener', {
+  port: 80,
+});
+
+const group1 = listener.addTargets('Target', {
+  port: 80,
+  targets: [new IpTarget('10.0.128.6')],
+});
+
+const group2 = listener.addTargets('ConditionalTarget', {
+  priority: 10,
+  hostHeader: 'example.com',
+  port: 80,
+  targets: [new IpTarget('10.0.128.5')],
+});
+
+vpc.publicSubnets.forEach((subnet, idx) => {
+  // Add a default ipv6 route to the subnet's route table.
+  const unboxedSubnet = subnet as ec2.Subnet;
+  unboxedSubnet.addRoute('IPv6Default', {
+    routerId: internetGateway.ref,
+    routerType: ec2.RouterType.GATEWAY,
+    destinationIpv6CidrBlock: '::/0',
+  });
+
+  // Find a CfnSubnet (raw cloudformation resources) child to the public
+  // subnet nodes.
+  const cfnSubnet = valueOrDie<IConstruct, ec2.CfnSubnet>(
+    subnet.node.children.find(c => c instanceof ec2.CfnSubnet),
+    new Error('Couldnt find a CfnSubnet'),
+  );
+
+  // Use the intrinsic Fn::Cidr CloudFormation function on the VPC's
+  // first IPv6 block to determine ipv6 /64 cidrs for each subnet as
+  // a function of the public subnet's index.
+  const vpcCidrBlock = cdk.Fn.select(0, vpc.vpcIpv6CidrBlocks);
+  const ipv6Cidrs = cdk.Fn.cidr(
+    vpcCidrBlock,
+    vpc.publicSubnets.length,
+    '64',
+  );
+  cfnSubnet.ipv6CidrBlock = cdk.Fn.select(idx, ipv6Cidrs);
+
+  // The subnet depends on the ipv6 cidr being allocated.
+  cfnSubnet.addDependency(ipv6Block);
+
+  group1.node.addDependency(subnet);
+  group2.node.addDependency(subnet);
+});
+
+listener.addAction('action1', {
+  priority: 1,
+  conditions: [
+    elbv2.ListenerCondition.hostHeaders(['example.com']),
+  ],
+  action: elbv2.ListenerAction.fixedResponse(200, { messageBody: 'success' }),
+});
+
+group1.metrics.targetResponseTime().createAlarm(stack, 'ResponseTimeHigh1', {
+  threshold: 5,
+  evaluationPeriods: 2,
+});
+
+group2.metrics.targetResponseTime().createAlarm(stack, 'ResponseTimeHigh2', {
+  threshold: 5,
+  evaluationPeriods: 2,
+});
+
+new integ.IntegTest(app, 'AlbDualstackWithoutPublicIpv4', {
+  testCases: [stack],
+});
+
+app.synth();
```

Comment:
```suggestion
```

you don't need `app.synth()` if your integ test uses `integ.IntegTest` (which it must, if it is a new integration test).